### DISCUSSION TITLE
apple data integration

### DISCUSSION THREAD

**Discussion by JBAnderson5** [almarklein/timetagger#152] [Ideas]
Hi Almarklein,

I just started using this app and I love how simple it is to use. 

I wanted to add some data points my apple devices are already tracking: namely workout and sleep data. Before I started coding, I figured I would ask you if anyone has done something similar or if this feature is already built into the product? I didn't find anything specific to apple in the docs. 

I also wanted to get your opinion on my development plan. I don't want to build an entire IOS app. I figured I could use shortcuts to access the healthkit api and read the data. That data is probably going to need to be processed into a different timetagger's format. Then I would use timetagger's web api to send the data to my server. Does that seem reasonable?

**Comment by almarklein** [comment]
Hi @JBAnderson5,

No, this functionality is not supported already. I would definitely recommend using TimeTagger's web API for this. You can then write a script or small application that collects the data you want, and send it to TimeTagger via the web API. That script/application needs to run on your machine, I imagine, and that way you keep it nice and modular, and it does not matter where the TimeTagger server is hosted.

Technically you could somehow make it part of the timetagger server process, but then it would (I think) need to run on localhost, you'd be forced to use Python, and it would be a bit more muddy to fit it in.

You can have a look at https://github.com/almarklein/timetagger_cli for a tool that uses the web api.

Feel free to ask if you need more help, this sounds very interesting!
